test/integration/framework/src/siftool/cosmos.py

```diff
@@ -19,10 +19,20 @@ def balance_normalize(bal: CompatBalance = None) -> Balance:
     return {k: v for k, v in bal.items() if v != 0}
 
 
-def balance_add(bal1: Balance, bal2: Balance) -> Balance:
+def balance_add(*bal: Balance) -> Balance:
     result = {}
-    for denom in set(bal1.keys()).union(set(bal2.keys())):
-        val = bal1.get(denom, 0) + bal2.get(denom, 0)
+    all_denoms = set(flatten_list([[*b.keys()] for b in bal]))
+    for denom in all_denoms:
+        val = sum(b.get(denom, 0) for b in bal)
+        if val != 0:
```

Comment:
I dont think you'll run into val == 0, It should be safe to remove this if clause

Reply:
Why not? `1 + (-1) == 0`. This keeps the balance dict normalized, which means it should not contain any keys for which the value is zero. In certain cases, test "== 0" could be a potential problem with numerical stability for float types, but AFAIK we're not doing anything like that for now.